client.tf

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  client_startup_script = "${file("${path.module}/scripts/forseti-client/forseti_client_startup_script.sh")}"
+  client_env_script     = "${file("${path.module}/scripts/forseti-client/forseti_environment.sh")}"
+  client_conf           = "${file("${path.module}/configs/forseti_conf_client.yaml")}"
+  client_conf_path      = "${var.forseti_home}/configs/forseti_conf_client.yaml"
+  client_sa_name        = "forseti-client-gcp-${local.random_hash}"
+  client_name           = "forseti-client-vm-${local.random_hash}"
+  client_zone           = "${var.client_region}-c"
```

Comment:
Another way to make the availability zone setting more flexible is to bring in a map of lists that includes all the possible zones you would want to deploy Forseti and rank sort by preference:

```
availability_zones = {
  "us-west1" = [
    "us-west1-a",
    "us-west1-b",
  ]
  "us-central1" = [
    "us-central1-b",
    "us-central1-f",
  ]
  "us-east1" = [
    "us-east1-c",
    "us-east1-d",
  ]
  "us-east4" = [
    "us-east4-a",
    "us-east4-b",
  ]
}
```
Once you have that data structure in place, you can do this in your locals block:
```
${element(var.availability_zones[var. client_region], 0)}
```

Reply:
Does this just help with validating if the zone is correct ? If it's not, we'll be out of that map and fail more gracefully ?

Reply:
Moved to https://github.com/terraform-google-modules/terraform-google-forseti/issues/18

Reply:
Follow up: how would someone choose us-west1-b if they're using `${element(var.availability_zones[var. client_region], var.client_zone_index)}` ? I don't like this client_zone_index which forces someone to look up the `availability_zones` map. Thoughts ?